packages/esm-outpatient-app/src/patient-search/patient-scheduled-visits.component.tsx

```diff
@@ -30,84 +31,100 @@ enum visitType {
-                          value={prioritySwitcherValue}
-                        />
-                        <Switch
-                          name={priority.PRIORITY}
-                          text={t('priority', 'Priority')}
-                          value={prioritySwitcherValue}
-                        />
-                        <Switch
-                          name={priority.EMERGENCY}
-                          text={t('emergency', 'Emergency')}
-                          value={prioritySwitcherValue}
-                        />
-                      </ContentSwitcher>
-                    ) : null}
-                  </div>
-                </RadioTile>
-              ))}
-            </TileGroup>
-          </div>
+                  {hasPriority && ind == visitsIndex ? (
+                    <ContentSwitcher
+                      size="sm"
+                      className={styles.prioritySwitcher}
+                      onChange={({ index }) => setSwitcherValue(index)}>
+                      <Switch
+                        name={priority.NOT_URGENT}
+                        text={t('notUrgent', 'Not Urgent')}
+                        value={prioritySwitcherValue}
+                      />
+                      <Switch name={priority.PRIORITY} text={t('priority', 'Priority')} value={prioritySwitcherValue} />
+                      <Switch
+                        name={priority.EMERGENCY}
+                        text={t('emergency', 'Emergency')}
+                        value={prioritySwitcherValue}
+                      />
+                    </ContentSwitcher>
+                  ) : null}
+                </div>
+              </RadioTile>
+            ))}
+          </TileGroup>
         ) : (
-          '--'
+          <div className={styles.emptyAppointment}>
+            <p>{t('noAppointmentsFound', 'No appointements found')} </p>
+          </div>
         )}
       </div>
     );
   }
+  return (
+    <div className={styles.emptyAppointment}>
+      <p className={styles.heading}> {scheduledVisitHeader} </p>
+    </div>
+  );
 };
 
 const PatientScheduledVisits: React.FC<PatientSearchProps> = ({ toggleSearchType, patientUuid }) => {
   const { t } = useTranslation();
   const isTablet = useLayoutType() === 'tablet';
-  const { recentVisits, isLoading } = useRecentScheduledVisits(patientUuid);
-  const { futureVisits, loading } = useFutureScheduledVisits(patientUuid);
+  const { appointments, isLoading, isError } = useScheduledVisits(patientUuid, new AbortController());
+
+  if (isError) {
+    <ErrorState headerTitle={t('errorFetchingAppoinments', 'Error fetching appointments')} error={isError} />;
+  }
+
+  if (isLoading) {
+    return <DataTableSkeleton role="progressbar" />;
+  }
+
+  if (
+    appointments === null ||
+    ((appointments.futureVisits === undefined || appointments.futureVisits === null) &&
+      (appointments.recentVisits === undefined || appointments.recentVisits === null))
```

Comment:
Will this work : ``` isNil(appointments.futureVisits) && isNil(appointments.recentVisits) ``` because im checking if there existing recent or fututre visits. And this is how appointments looks ``` appointments: data ? { recentVisits, futureVisits } : null ```